node/src/database/db_migrations.rs

```diff
@@ -74,353 +158,684 @@ impl DbMigratorReal {
+        transaction.execute(
+            &format!(
+                "UPDATE {} SET value = ? WHERE name = 'schema_version'",
+                name_of_given_table
+            ),
+            &[updated_to],
+        )?;
         Ok(())
     }
 
-    fn aggregated_checks<'a>(
-        mismatched_schema: &str,
-        list_of_updates: &'a [&'a (dyn MigrateDatabase + 'a)],
-    ) -> Result<Vec<&'a (dyn MigrateDatabase + 'a)>, String> {
-        Self::schema_initial_validation_check(mismatched_schema);
-        let updates_to_process = Self::list_validation_check(list_of_updates)
-            .skip_while(|entry| entry.version_compatibility().ne(mismatched_schema))
-            .map(|e| e.deref())
-            .collect::<Vec<&(dyn MigrateDatabase + 'a)>>();
-        let _ =
-            Self::check_the_number_of_those_remaining(mismatched_schema, updates_to_process.len())?;
-        Ok(updates_to_process)
-    }
-
-    fn process_items_from_beneath_dirty_references<'a>(
-        first: Option<&'a &dyn MigrateDatabase>,
-        second: Option<&&&dyn MigrateDatabase>,
-    ) -> (&'a dyn MigrateDatabase, String) {
-        let first = *first.expect_decent("migration record");
-        let second_by_its_name = if let Some(next) = second {
-            next.version_compatibility()
+    fn is_target_version_reached(current_state: usize, target_version: &usize) -> bool {
+        if current_state.lt(target_version) {
+            false
+        } else if current_state.eq(target_version) {
+            true
         } else {
-            CURRENT_SCHEMA_VERSION
-        };
-        (
-            first,
-            Self::identification_of_versions(first.version_compatibility(), second_by_its_name),
-        )
+            panic!("Nonsense: the given target is lower than the version that is considered mismatched")
+        }
     }
 
-    fn check_the_number_of_those_remaining(
-        mismatched_schema: &str,
-        count: usize,
-    ) -> Result<(), String> {
-        match count {
-            0 => Err(format!("Database claims to be more advanced ({}) than the version {} which is the latest released.", mismatched_schema, CURRENT_SCHEMA_VERSION)),
-            _ => Ok(())
+    fn select_updates_to_process<'a>(
+        mismatched_schema: usize,
+        list_of_updates: &'a [&'a (dyn DatabaseMigration + 'a)],
+    ) -> Vec<&'a (dyn DatabaseMigration + 'a)> {
+        let updates_to_process = list_of_updates
+            .iter()
+            .skip_while(|entry| entry.old_version().ne(&mismatched_schema))
+            .map(Self::deref)
+            .collect::<Vec<&(dyn DatabaseMigration + 'a)>>();
+        let _ = Self::check_out_quantity_of_those_remaining(
+            mismatched_schema,
+            updates_to_process.len(),
+        );
+        updates_to_process
+    }
+
+    fn deref<'a>(value: &'a &dyn DatabaseMigration) -> &'a dyn DatabaseMigration {
+        *value
+    }
+
+    fn process_items_with_dirty_references<'a>(
```

Comment:
There are some pretty unclear names here in these two functions, starting with "process_items_with_dirty_references." In software development, the term "dirty" has a technical meaning that usually has to do with changes in memory that have not yet been written out to somewhere, like disk or network or display. It seems here to mean "burdened with lots of levels of references," which isn't the same thing. Maybe a different word could be used? Also, "first" and "second" might be changed to "from" and "to," since "first" isn't really version 0 and "second" isn't really version 1.